From 7ecbf26da075691e1ccc3e9d5476db812ddeeede Mon Sep 17 00:00:00 2001
From: Stephane Viau <stephane.viau@nxp.com>
Date: Mon, 8 Jan 2024 13:02:09 +0100
Subject: [PATCH 10/28] configs: arm64: imx8m{m,n,p}: add harpoon freertos
 industrial cell

Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
(rename configs: add harpoon prefix)
Signed-off-by: Marouen Ghodhbane <marouen.ghodhbane@nxp.com>
(imx8mp: freertos: Add FlexCAN1 to the industrial cell config)
Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
(imx8mp: freertos: Add ENET_QOS to the industrial cell config)
Signed-off-by: Stoica Cosmin-Stefan <cosmin.stoica@nxp.com>
(configs: imx8m{m,n}: freertos: Add resources for ENET to industrial cell)
Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
---
 .../imx8mm-harpoon-freertos-industrial.c      | 62 ++++++++++++++++++
 configs/arm64/imx8mm-harpoon-rtos.h           |  4 ++
 .../imx8mn-harpoon-freertos-industrial.c      | 62 ++++++++++++++++++
 configs/arm64/imx8mn-harpoon-rtos.h           |  4 ++
 .../imx8mp-harpoon-freertos-industrial.c      | 65 +++++++++++++++++++
 configs/arm64/imx8mp-harpoon-rtos.h           |  5 ++
 6 files changed, 202 insertions(+)
 create mode 100644 configs/arm64/imx8mm-harpoon-freertos-industrial.c
 create mode 100644 configs/arm64/imx8mn-harpoon-freertos-industrial.c
 create mode 100644 configs/arm64/imx8mp-harpoon-freertos-industrial.c

diff --git a/configs/arm64/imx8mm-harpoon-freertos-industrial.c b/configs/arm64/imx8mm-harpoon-freertos-industrial.c
new file mode 100644
index 00000000..dbe7fda1
--- /dev/null
+++ b/configs/arm64/imx8mm-harpoon-freertos-industrial.c
@@ -0,0 +1,62 @@
+/*
+ * i.MX8MM target - FreeRTOS industrial config
+ *
+ * Copyright 2022 NXP
+ *
+ * Authors:
+ *  Stephane Viau <stephane.viau@nxp.com>
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2.  See
+ * the COPYING file in the top-level directory.
+ */
+
+#include "cell-helper.h"
+#include "imx8mm-harpoon-rtos.h"
+
+/* Name, cores, entry point */
+#define CONFIG_INMATE_NAME		"freertos"
+#define CONFIG_INMATE_CORE_BITMAP	(0b1000)
+#define CONFIG_INMATE_BASE		(0xc0000000)
+
+/* Memory & peripherals */
+#define CONFIG_INMATE_REGIONS		\
+	RTOS_COMMON_MEMORY_REGIONS,                                     \
+	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */    \
+	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
+	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
+	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
+	MMIO_REGION_RWS(0x30360000, 0x30360000, KB(64)),   /* ANA_PLL */\
+	MMIO_REGION_RWS(0x30380000, 0x30380000, KB(64)),   /* CCM */    \
+	MMIO_REGION_RW( 0x30be0000, 0x30be0000, KB(4))     /* ENET */
+
+/* GIC */
+#define CONFIG_INMATE_IRQCHIPS_ADDR	(0x38800000)
+#define CONFIG_INMATE_IRQCHIPS \
+	IRQCHIP0(CONFIG_INMATE_IRQCHIPS_ADDR,              \
+		/* interrupts 32..63 */                    \
+		RTOS_COMMON_IRQCHIP0_BITMAP1,              \
+		/* interrupts 64..95 */                    \
+		RTOS_COMMON_IRQCHIPS_BITMAP2,              \
+		/* interrupts 96..127 */                   \
+		RTOS_COMMON_IRQCHIPS_BITMAP3,              \
+		/* interrupts 128..159 */                  \
+		RTOS_COMMON_IRQCHIPS_BITMAP4        |      \
+		IRQnMAP(ENET_MAC0_Rx_Tx_Done1_IRQn) |      \
+		IRQnMAP(ENET_MAC0_Rx_Tx_Done2_IRQn) |      \
+		IRQnMAP(ENET_IRQn)                  |      \
+		IRQnMAP(ENET_1588_IRQn)),                  \
+	IRQCHIP1(CONFIG_INMATE_IRQCHIPS_ADDR,              \
+		/* interrupts 160..191 */                  \
+		0,                                         \
+		/* interrupts 192..223 */                  \
+		0,                                         \
+		/* interrupts 224..255 */                  \
+		0,                                         \
+		/* interrupts 256..287 */                  \
+		0)
+
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
+#include "cell-create.h"
diff --git a/configs/arm64/imx8mm-harpoon-rtos.h b/configs/arm64/imx8mm-harpoon-rtos.h
index 5bb54cf0..4acd1322 100644
--- a/configs/arm64/imx8mm-harpoon-rtos.h
+++ b/configs/arm64/imx8mm-harpoon-rtos.h
@@ -20,6 +20,10 @@ typedef enum IRQn {
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
+  ENET_MAC0_Rx_Tx_Done1_IRQn   = 150,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
+  ENET_MAC0_Rx_Tx_Done2_IRQn   = 151,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
+  ENET_IRQn                    = 152,              /**< MAC 0 IRQ */
+  ENET_1588_IRQn               = 153,              /**< MAC 0 1588 Timer Interrupt - synchronous */
 } IRQn_Type;
 
 #define RTOS_COMMON_MEMORY_REGIONS \
diff --git a/configs/arm64/imx8mn-harpoon-freertos-industrial.c b/configs/arm64/imx8mn-harpoon-freertos-industrial.c
new file mode 100644
index 00000000..e514dbd8
--- /dev/null
+++ b/configs/arm64/imx8mn-harpoon-freertos-industrial.c
@@ -0,0 +1,62 @@
+/*
+ * i.MX8MN target - FreeRTOS industrial config
+ *
+ * Copyright 2022 NXP
+ *
+ * Authors:
+ *  Stephane Viau <stephane.viau@nxp.com>
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2.  See
+ * the COPYING file in the top-level directory.
+ */
+
+#include "cell-helper.h"
+#include "imx8mn-harpoon-rtos.h"
+
+/* Name, cores, entry point */
+#define CONFIG_INMATE_NAME		"freertos"
+#define CONFIG_INMATE_CORE_BITMAP	(0b1000)
+#define CONFIG_INMATE_BASE		(0xc0000000)
+
+/* Memory & peripherals */
+#define CONFIG_INMATE_REGIONS		\
+	RTOS_COMMON_MEMORY_REGIONS,                                     \
+	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */    \
+	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
+	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
+	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
+	MMIO_REGION_RWS(0x30360000, 0x30360000, KB(64)),   /* ANA_PLL */\
+	MMIO_REGION_RWS(0x30380000, 0x30380000, KB(64)),   /* CCM */    \
+	MMIO_REGION_RW( 0x30be0000, 0x30be0000, KB(4))     /* ENET */
+
+/* GIC */
+#define CONFIG_INMATE_IRQCHIPS_ADDR	(0x38800000)
+#define CONFIG_INMATE_IRQCHIPS \
+	IRQCHIP0(CONFIG_INMATE_IRQCHIPS_ADDR,              \
+		/* interrupts 32..63 */                    \
+		RTOS_COMMON_IRQCHIP0_BITMAP1,              \
+		/* interrupts 64..95 */                    \
+		RTOS_COMMON_IRQCHIPS_BITMAP2,              \
+		/* interrupts 96..127 */                   \
+		RTOS_COMMON_IRQCHIPS_BITMAP3,              \
+		/* interrupts 128..159 */                  \
+		RTOS_COMMON_IRQCHIPS_BITMAP4        |      \
+		IRQnMAP(ENET_MAC0_Rx_Tx_Done1_IRQn) |      \
+		IRQnMAP(ENET_MAC0_Rx_Tx_Done2_IRQn) |      \
+		IRQnMAP(ENET_IRQn)                  |      \
+		IRQnMAP(ENET_1588_IRQn)),                  \
+	IRQCHIP1(CONFIG_INMATE_IRQCHIPS_ADDR,              \
+		/* interrupts 160..191 */                  \
+		0,                                         \
+		/* interrupts 192..223 */                  \
+		0,                                         \
+		/* interrupts 224..255 */                  \
+		0,                                         \
+		/* interrupts 256..287 */                  \
+		0)
+
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
+#include "cell-create.h"
diff --git a/configs/arm64/imx8mn-harpoon-rtos.h b/configs/arm64/imx8mn-harpoon-rtos.h
index 05300e95..dca81931 100644
--- a/configs/arm64/imx8mn-harpoon-rtos.h
+++ b/configs/arm64/imx8mn-harpoon-rtos.h
@@ -21,6 +21,10 @@ typedef enum IRQn {
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
+  ENET_MAC0_Rx_Tx_Done1_IRQn   = 150,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
+  ENET_MAC0_Rx_Tx_Done2_IRQn   = 151,              /**< MAC 0 Receive / Trasmit Frame / Buffer Done */
+  ENET_IRQn                    = 152,              /**< MAC 0 IRQ */
+  ENET_1588_IRQn               = 153,              /**< MAC 0 1588 Timer Interrupt - synchronous */
 } IRQn_Type;
 
 #define RTOS_COMMON_MEMORY_REGIONS \
diff --git a/configs/arm64/imx8mp-harpoon-freertos-industrial.c b/configs/arm64/imx8mp-harpoon-freertos-industrial.c
new file mode 100644
index 00000000..fc94acbe
--- /dev/null
+++ b/configs/arm64/imx8mp-harpoon-freertos-industrial.c
@@ -0,0 +1,65 @@
+/*
+ * i.MX8MP target - FreeRTOS industrial config
+ *
+ * Copyright 2022 NXP
+ *
+ * Authors:
+ *  Stephane Viau <stephane.viau@nxp.com>
+ *
+ * This work is licensed under the terms of the GNU GPL, version 2.  See
+ * the COPYING file in the top-level directory.
+ */
+
+#include "cell-helper.h"
+#include "imx8mp-harpoon-rtos.h"
+
+/* Name, cores, entry point */
+#define CONFIG_INMATE_NAME		"freertos"
+#define CONFIG_INMATE_CORE_BITMAP	(0b1000)
+#define CONFIG_INMATE_BASE		(0xc0000000)
+
+/* Memory & peripherals */
+#define CONFIG_INMATE_REGIONS		\
+	RTOS_COMMON_MEMORY_REGIONS,                                     \
+	MEM_REGION_RWXL(0xc0000000, 0xc0000000, MB(16)),   /* RAM */    \
+	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
+	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
+	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
+	MMIO_REGION_RWS(0x30230000, 0x30230000, KB(64)),   /* GPIO4 */  \
+	MMIO_REGION_RWS(0x30240000, 0x30240000, KB(64)),   /* GPIO5 */  \
+	MMIO_REGION_RWS(0x30340000, 0x30340000, KB(64)),   /* GPR */    \
+	MMIO_REGION_RWS(0x30360000, 0x30360000, KB(4)),    /* ANA_PLL */\
+	MMIO_REGION_RWS(0x30380000, 0x30380000, KB(64)),   /* CCM */    \
+	MMIO_REGION_RW( 0x308c0000, 0x308c0000, KB(64)),   /* FLEXCAN1*/\
+	MMIO_REGION_RW( 0x30bf0000, 0x30bf0000, KB(64))    /* ENET_Q */
+
+/* GIC */
+#define CONFIG_INMATE_IRQCHIPS_ADDR	(0x38800000)
+#define CONFIG_INMATE_IRQCHIPS \
+	IRQCHIP0(CONFIG_INMATE_IRQCHIPS_ADDR,              \
+		/* interrupts 32..63 */                    \
+		RTOS_COMMON_IRQCHIP0_BITMAP1,              \
+		/* interrupts 64..95 */                    \
+		RTOS_COMMON_IRQCHIPS_BITMAP2,              \
+		/* interrupts 96..127 */                   \
+		RTOS_COMMON_IRQCHIPS_BITMAP3,              \
+		/* interrupts 128..159 */                  \
+		RTOS_COMMON_IRQCHIPS_BITMAP4),             \
+	IRQCHIP1(CONFIG_INMATE_IRQCHIPS_ADDR,              \
+		/* interrupts 160..191 */                  \
+		IRQnMAP(ENET_QOS_PMT_IRQn) |               \
+		IRQnMAP(ENET_QOS_IRQn) |                   \
+		IRQnMAP(CAN_FD1_IRQn) | /* FLEXCAN1 */     \
+		IRQnMAP(CAN_FD1_ERROR_IRQn),               \
+		/* interrupts 192..223 */                  \
+		0,                                         \
+		/* interrupts 224..255 */                  \
+		0,                                         \
+		/* interrupts 256..287 */                  \
+		0)
+
+/* IVSHMEM PCI */
+#define CONFIG_INMATE_PCI_DEVICES \
+	RTOS_COMMON_PCI_IVSHMEM
+
+#include "cell-create.h"
diff --git a/configs/arm64/imx8mp-harpoon-rtos.h b/configs/arm64/imx8mp-harpoon-rtos.h
index ed343c5a..47cb99f1 100644
--- a/configs/arm64/imx8mp-harpoon-rtos.h
+++ b/configs/arm64/imx8mp-harpoon-rtos.h
@@ -20,6 +20,10 @@ typedef enum IRQn {
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
+  ENET_QOS_PMT_IRQn            = 166,              /**< ENET QOS TSN Interrupt from PMT */
+  ENET_QOS_IRQn                = 167,              /**< ENET QOS TSN LPI RX exit/Host System/RX/TX Channels[4:0] Interrupt */
+  CAN_FD1_IRQn                 = 174,              /**< CAN-FD1 Interrupt from bus off/line error/RX warning/TX warning/wakeup/match in PN/timeout in PN/busoff done/FD error */
+  CAN_FD1_ERROR_IRQn           = 175,              /**< CAN-FD1 Interrupt from correctable error/non correctable error int host/ non correctable error int internal */
 } IRQn_Type;
 
 #define RTOS_COMMON_MEMORY_REGIONS \
@@ -29,6 +33,7 @@ typedef enum IRQn {
 	MEM_REGION_RWS( 0xfd9fc000, 0xfd9fc000, KB(8)),  /* IVSHMEM */              \
 	MEM_REGION_ROS( 0xfd9fe000, 0xfd9fe000, KB(8)),  /* IVSHMEM */              \
 	/* IVSHMEM regions should always be kept on top so PCI_IVS doesn't change */\
+	MMIO_REGION_ROS(0x30270000, 0x30270000, KB(64)), /* ANA_OSC */              \
 	MMIO_REGION_RW( 0x302d0000, 0x302d0000, KB(64)), /* GPT1 */                 \
 	MMIO_REGION_RW( 0x302e0000, 0x302e0000, KB(64)), /* GPT2 */                 \
 	MMIO_REGION_RWS(0x30330000, 0x30330000, KB(64)), /* IOMUXC */               \
-- 
2.34.1

